src/sinks/loki.rs

```diff
@@ -289,11 +289,35 @@ async fn healthcheck(config: LokiConfig, client: HttpClient) -> crate::Result<()
 
     let res = client.send(req).await?;
 
-    if res.status() != http::StatusCode::OK {
-        return Err(format!("A non-successful status returned: {}", res.status()).into());
+    let status = match fetch_status("ready", &config, &client).await? {
+        // Issue https://github.com/timberio/vector/issues/6463
+        http::StatusCode::NOT_FOUND => {
+            warn!("Endpoint `/ready` not found. Retrying healthcheck with top level query.");
```

Comment:
Given this will be the common case when using Grafana Cloud, I think I might argue to reduce this to `debug!` to avoid spamming the logs.